contracts/staking/PoSValidatorManager.sol

```diff
@@ -88,16 +92,26 @@ abstract contract PoSValidatorManager is IPoSValidatorManager, ValidatorManager
         bool includeUptimeProof,
         uint32 messageIndex
     ) external {
+        uint64 uptime;
         if (includeUptimeProof) {
-            _getUptime(validationID, messageIndex);
+            uptime = _getUptime(validationID, messageIndex);
         }
-        // TODO: Calculate the reward for the validator, but do not unlock it
 
-        _initializeEndValidation(validationID);
+        Validator memory validator = _initializeEndValidation(validationID);
+
+        PoSValidatorManagerStorage storage $ = _getPoSValidatorManagerStorage();
+        $._pendingValidatorRewards[validator.owner] +=
+            $._rewardCalculator.calculateReward(validator.weight, uptime, 0, 0);
```

Comment:
this needs to be calculated with `validator.startingWeight`, since `weight` includes all the weight of the delegations, which should not be given to the validator.